google.com/src/ada_google_com/sheets.py

```diff
@@ -1,133 +1,239 @@
+        <Raw Content>
+        <Foot: See Foot Implementation>
+        ------------------------------------------------------------------------
+
+    """
+    spreadsheet_id: str
+    sheet_name: str
+    raw_content: Sequence[Sequence[str]]
+    head_index: HeadIndex
+    foot: Foot
+
+    @classmethod
+    def from_config(cls,
+                    spreadsheet_id: str,
+                    sheets: Sequence[str],
+                    credentials: Credentials) -> Iterable['Source']:
+        """
+        Factory method to create a Source object, by:
+
+        - fetch sheet info form google api on spreadsheet_id and sheet_name
+        - extract header, content and footer
+        """
+        service = build('sheets', 'v4', credentials=credentials)
+
+        # Call the Sheets API
+        sheet = service.spreadsheets()  # pylint: disable=no-member
+        output = []
+        for sheet_name in sheets:
+            output.append('INFO Fetching Source(%r, %r)' % (  # pylint: disable=consider-using-f-string
+                spreadsheet_id, sheet_name
+            ))
+
+            fetch_result = cls._fetch(
+                sheet=sheet,
+                spreadsheet_id=spreadsheet_id,
+                sheet_name=sheet_name
+            )
+
+            raw_head, *raw_content, raw_footer = fetch_result.get('values', [])
+
+            head_index = HeadIndex.from_sheet_head(
+                sheet_name=sheet_name,
+                head=raw_head
+            )
+            foot = Foot.from_sheet_foot_and_head_index(
+                foot=raw_footer,
+                head_index=head_index,
+            )
+
+            yield Source(
+                spreadsheet_id=spreadsheet_id,
+                sheet_name=sheet_name,
+                raw_content=raw_content,
+                head_index=head_index,
+                foot=foot
+            )
+
+    def _process_content_single_line(self, row: Sequence[str]) -> Optional[str]:
+        now = datetime.utcnow()
+
+        last_date_checked_in = row[self.head_index.last_time_logged_in]
         if last_date_checked_in == 'skip':
-            continue
+            return None
 
-        name = row[name_index]
-        user = row[user_index]
-        email = row[email_index]
+        name = row[self.head_index.name]
+        user = row[self.head_index.user]
+        email = row[self.head_index.email]
 
         date = datetime.strptime(last_date_checked_in, '%Y.%m.%d')
-        if (now - date) > timedelta(days=notify_after_n_days):
-            need_notification = True
-            output.append((
-                'you have to log into to refresh account, '
-                f'you could loose that account .... {repr(name)!r}, {user!r}, {email!r}')
+        if (now - date) > timedelta(days=self.foot.notify_after_n_days):
+            return (
+                'You have to log into to refresh account, '
+                f'you could loose that account .... {repr(name)!r}, {user!r}, {email!r}'
```

Comment:
loose -> lose

Reply:
Also repr(name) is unnecessary if **!r** flag is used in f-string

Reply:
fixed.